rd_server.py

```diff
@@ -322,7 +322,7 @@ def scheduler_tick():
                     rd_print('Failed to write results for work item',slot.work.get_name())
                 work_done.append(slot.work)
                 rd_print(slot.work.log,slot.work.get_name(),'finished.')
-            elif slot.work.retries < max_retries and not slot.work.run.cancelled and not slot.p.returncode == 98:
+            elif slot.work.retries < max_retries and not slot.work.run.cancelled and (slot.p.p == None or not slot.p.p.returncode == 98):
```

Comment:
nit: `is None` is more idiomatic than `== None`. 

Also (and this is subjective) I find `or slot.p.p.returncode != 98` more readable than `or not slot.p.p.returncode == 98` but feel free to disregard.